### PR TITLE
[VEN-2227]: Add MoveDebt delegate borrower

### PR DESCRIPTION
**Pull request by kkirka**
## Description

This PR adds `MoveDebtDelegate` – a delegate borrower contract that

* repays debt of one user in a preconfigured market
* borrows on behalf of another user in any market

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [x] If I added new functionality, I added tests covering it.
- [x] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [x] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/DelegateBorrowers/MoveDebtDelegate.sol [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+     * @param originalBorrower The address of the borrower, whose debt to repay
+     * @param repayAmount The amount to repay in terms of vTokenToRepay.underlying()
+     * @param newBorrower The address of the borrower, who will own the new borrow
+     * @param vTokenToBorrow VToken to borrow from
+     */
+    function moveDebt(
+        address originalBorrower,
+        uint256 repayAmount,
+        address newBorrower,
+        IVBep20 vTokenToBorrow
```
Comment (chechu) [inline contracts/DelegateBorrowers/MoveDebtDelegate.sol:72]: We could check this `vTokenToBorrow` is in one pre-fixed list of vtokens, like vUSDT and vUSDC

File: contracts/DelegateBorrowers/MoveDebtDelegate.sol [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    ) external onlyOwner nonReentrant {
+        uint256 actualRepaymentAmount = _repay(vTokenToRepay, originalBorrower, repayAmount);
+        uint256 amountToBorrow = _convert(vTokenToRepay, vTokenToBorrow, actualRepaymentAmount);
```
[the remark this answers is not in the document]
Comment (kkirka) [reply contracts/DelegateBorrowers/MoveDebtDelegate.sol]: I've removed the param from `_repay` but I'd rather keep vTokens as parameters in `_convert` for readability.

File: contracts/DelegateBorrowers/MoveDebtDelegate.sol (hunk 1 above)
Comment (kkirka) [reply contracts/DelegateBorrowers/MoveDebtDelegate.sol]: [da6a48f](https://github.com/VenusProtocol/venus-protocol/pull/406/commits/da6a48f44e088bc73599c2abbe5a001409509d58)

**Review by chechu — APPROVED** [review]

File: contracts/DelegateBorrowers/MoveDebtDelegate.sol [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    error BorrowFailed(uint256 errorCode);
+
+    /// @notice Thrown if borrowing from the corresponding vToken is not allowed
+    error BorrowNotAllowed(address vToken);
+
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    /// @notice Constructor for the implementation contract. Sets immutable variables.
+    /// @param vTokenToRepay_ VToken to repay the debt to
+    /// @notice User to borrow on behalf of
```
[the remark this answers is not in the document]
Comment (kkirka) [reply contracts/DelegateBorrowers/MoveDebtDelegate.sol]: [a8b6789](https://github.com/VenusProtocol/venus-protocol/pull/406/commits/a8b6789d5af674d94fc1c3236b91f04025b25cdb)

**Review by chechu — APPROVED** [review]

**Pull request merged**